Library/Tracking/KSRAnalytics.swift

```diff
@@ -379,6 +379,26 @@ public final class KSRAnalytics {
     case watch
     case watched
 
+    /**
+     Initialize `TypeContext` with `DiscoveryParams`
+     to determine type context for discovery collection filters.
+
+     - parameter params: a `DiscoveryParams` object
+     */
+    init?(params: DiscoveryParams) {
+      if let recommended = params.recommended, recommended {
+        self = .recommended
+      } else if let starred = params.starred, starred {
+        self = .watched
+      } else if let staffPicks = params.staffPicks, staffPicks {
+        self = .pwl
+      } else if let social = params.social, social {
+        self = .social
+      } else {
+        return nil
```

Comment:
this should also be `self = .results`. see if you can consolidate the logic for both cases that require a `self = .results`